aspnet/mvc/views/razor.rst

```diff
@@ -1,8 +1,732 @@
+
+.. code-block:: none
+
+  @using (Html.BeginForm())
+  {
+      // Form content.
+  }
+
+You can also perform scope level actions like the above with :doc:`/mvc/views/tag-helpers/index`.
+
+``@try``, ``catch``, ``finally``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Exception handling is similar to  C#:
+
+.. literalinclude:: razor/sample/Views/Home/Contact7.cshtml
+  :language: html
+
+``@lock``
+^^^^^^^^^
+
+Razor has the capability to protect critical sections with lock statements:
+
+.. code-block:: none
+
+  @lock (SomeLock)
+  {
+      // Do critical section work
+  }
+
+Comments
+^^^^^^^^^^
+
+Razor supports C# and HTML comments. The following markup:
+
+.. code-block:: none
+
+  @{
+      /* C# comment. */
+      // Another C# comment.
+  }
+  <!-- HTML comment -->
+
+Is rendered by the server as:
+
+.. code-block:: none
+
+  <!-- HTML comment -->
+
+Razor comments are removed by the server before the page is rendered. Razor uses ``@*  *@`` to delimit comments. The following code is commented out, so the server will not render any markup:
+
+.. code-block:: none
+
+    @*
+    @{
+        /* C# comment. */
+        // Another C# comment.
+    }
+    <!-- HTML comment -->
+   *@
+
+.. _razor-directives-label:
+
+Directives
+-----------
+Razor directives are represented by implicit expressions with reserved keywords following the ``@`` symbol. A directive will typically change the way a page is parsed or enable different functionality within your Razor page. A Razor page is just a generated C# file. A simple example of what a Razor page generates:
+
+.. literalinclude:: razor/sample/Views/Home/Contact8.cshtml
+  :language: html
+
+The Razor markup above will generate a class similar to the following:
+
+.. code-block:: c#
+
+  public class _Views_Something_cshtml : RazorPage<dynamic>
+  {
+      public override async Task ExecuteAsync()
+      {
+          var output = "Hello World";
+
+          WriteLiteral("/r/n<div>Output: ");
+          Write(output);
+          WriteLiteral("</div>");
+      }
+  }
+
+:ref:`razor-customcompilationservice-label` explains how to view this generated class. Understanding how Razor generates code for a view will make it easier to understand how directives work.
+
+``@using``
+^^^^^^^^^^^^^
+
+The ``@using`` directive will add the c# ``using`` directive to the generated razor page:
+
+.. literalinclude:: razor/sample/Views/Home/Contact9.cshtml
+  :language: html
+
+``@model``
+^^^^^^^^^^^^
+
+The ``@model`` directive allows you to specify the type of the model passed to your Razor page. It uses the following syntax:
+
+.. code-block:: none
+
+  @model TypeNameOfModel<TModel>
+
+For example, if you create a new ASP.NET Core MVC app with individual user accounts, the *Views/Account/Login.cshtml* Razor view file contains the follow model declaration:
```

Comment:
I suggest replacing the rest of this section with:

Razor pages expose a `Model` property for accessing the model passed to the page.  

``` rst
<div>The Login Email: @Model.Email</div>
```

The `@model` directive specified the type of this property (by specifying the `T` in `RazorPage<T>` that the generated class for your page derives from). If you don't specify the `@model` directive the `Model` property will be of type `dynamic`. The value of the model is passed from the controller to the view. See `` :ref:`strongly-typed-models-keyword-label` `` for more information.